pkg/docker/config/config.go

```diff
@@ -55,7 +56,27 @@ var (
 
 // SetAuthentication stores the username and password in the auth.json file
 func SetAuthentication(sys *types.SystemContext, registry, username, password string) error {
+	credConfigs, err := sysregistriesv2.GetRegistryCredentialHelpers(sys, registry)
+	if err != nil {
+		return err
+	}
+	if len(credConfigs) > 0 {
+		if err := func(credConfigs []string, registry, username, password string) error {
+			for _, credConfig := range credConfigs {
```

Comment:
```suggestion
			for _, helper := range helpers {
```
That’s what it is, isn’t it? There isn’t any other configuration.